pywinauto/hooks.py

```diff
@@ -0,0 +1,317 @@
+
+        if self.keyboard_is_hook:
+            def keyboard_low_level_handler(code, event_code, kb_data_ptr):
+                """Execute when keyboard low level event was catched"""
+                try:
+                    key_code = 0xFFFFFFFF & kb_data_ptr[0]
+                    current_key = self.ID_TO_KEY[key_code]
+                    event_type = self.event_types[0xFFFFFFFF & event_code]
+
+                    if event_type == 'key down':
+                        self.pressed_keys.append(current_key)
+
+                    if event_type == 'key up':
+                        self.pressed_keys.remove(current_key)
+
+                    event = KeyboardEvent(current_key, event_type, self.pressed_keys)
+
+                    if self.handler != 0:
+                        self.handler(event)
+
+                finally:
+                    return windll.user32.CallNextHookEx(self.keyboard_id, code, event_code, kb_data_ptr)
+
+            keyboard_pointer = _callback_pointer(keyboard_low_level_handler)
+
+            self.keyboard_id = windll.user32.SetWindowsHookExA(self.WH_KEYBOARD_LL, keyboard_pointer,
+                                                               windll.kernel32.GetModuleHandleA(None),
+                                                               0)
+
+        if self.mouse_is_hook:
+            def mouse_low_level_handler(code, event_code, kb_data_ptr):
+                """Execute when mouse low level event was catched"""
+                try:
+                    current_key = self.MOUSE_ID_TO_KEY[event_code]
+                    if current_key != 'Move':
+                        event_type = self.MOUSE_ID_TO_EVENT_TYPE[event_code]
+                        event = MouseEvent(current_key, event_type)
+
+                        if self.handler != 0:
+                            self.handler(event)
+
+                finally:
+                    return windll.user32.CallNextHookEx(self.mouse_id, code, event_code, kb_data_ptr)
+
+            mouse_pointer = _callback_pointer(mouse_low_level_handler)
+            self.mouse_id = windll.user32.SetWindowsHookExA(self.WH_MOUSE_LL, mouse_pointer,
+                                                            windll.kernel32.GetModuleHandleA(None), 0)
+
+        self.listen()
+
+    def unhook_mouse(self):
+        """Unhook mouse events"""
+        self.mouse_is_hook = False
+        windll.user32.UnhookWindowsHookEx(self.mouse_id)
+
+    def unhook_keyboard(self):
+        """Unhook keyboard events"""
+        self.keyboard_is_hook = False
+        windll.user32.UnhookWindowsHookEx(self.keyboard_id)
+
+    def listen(self):
+        """Listen events"""
+        atexit.register(windll.user32.UnhookWindowsHookEx, self.keyboard_id)
+        atexit.register(windll.user32.UnhookWindowsHookEx, self.mouse_id)
+
+        message_pointer = MSG()
+
+        while self.mouse_is_hook or self.keyboard_is_hook:
+            msg = windll.user32.GetMessageW(message_pointer, 0, 0, 0)
```

Comment:
@MaxSamokhvalov you don't check the return value. Did you read the MSDN about [GetMessage](https://msdn.microsoft.com/en-us/library/windows/desktop/ms644936(v=vs.85).aspx) that I pointed out ? I'd still consider to use non-blocking functions. For example, using [PeekMessage](https://msdn.microsoft.com/en-us/library/windows/desktop/ms644943(v=vs.85).aspx) will allow to be more responsive with more frequent checking the main condition ```while self.mouse_is_hook or self.keyboard_is_hook```